reconcile/utils/terrascript_client.py

```diff
@@ -603,6 +608,54 @@ def populate_tgw_attachments(self, desired_state):
                 aws_ram_resource_association(identifier, **values)
             self.add_resource(req_account_name, tf_resource_association)
 
+            # now that the tgw is shared to the cluster's aws account
+            # we can create a vpc attachment to the tgw
+            subnets_id_az = accepter['subnets_id_az']
+            subnets = self.get_az_unique_subnet_ids(subnets_id_az)
+            values = {
+                'provider': 'aws.' + acc_alias,
+                'subnet_ids': subnets,
+                'transit_gateway_id': requester['tgw_id'],
+                'vpc_id': accepter['vpc_id'],
+                'depends_on': [
+                    'aws_ram_principal_association.' + connection_name,
+                    'aws_ram_resource_association.' + identifier
+                ],
+                'tags': tags
+            }
+            tf_resource_attachment = \
+                aws_ec2_transit_gateway_vpc_attachment(identifier, **values)
+            # we send the attachment from the cluster's aws account
+            self.add_resource(acc_account_name, tf_resource_attachment)
+
+            # and accept the attachment in the non cluster's aws account
+            values = {
+                'transit_gateway_attachment_id':
+                    '${' + tf_resource_attachment.id + '}',
+                'tags': tags
+            }
+            if self._multiregion_account_(req_account_name):
+                values['provider'] = 'aws.' + requester['region']
+            tf_resource_attachment_accepter = \
+                aws_ec2_transit_gateway_vpc_attachment_accepter(
+                    identifier, **values)
+            self.add_resource(
+                req_account_name, tf_resource_attachment_accepter)
+
+    @staticmethod
+    def get_az_unique_subnet_ids(subnets_id_az):
```

Comment:
what's `subnets_id_az` exactly?

Reply:
it's a list of items with a subnet's id and availability zone. it is needed later to add subnets to the VPC attachment (and only 1 per AZ is allowed, hence we are passing the AZ).